ambry-frontend/src/main/java/com/github/ambry/frontend/GetDatasetsHandler.java

```diff
@@ -0,0 +1,161 @@
+import com.github.ambry.account.AccountService;
+import com.github.ambry.account.AccountServiceException;
+import com.github.ambry.account.Dataset;
+import com.github.ambry.commons.ByteBufferReadableStreamChannel;
+import com.github.ambry.commons.Callback;
+import com.github.ambry.rest.RestRequest;
+import com.github.ambry.rest.RestRequestMetrics;
+import com.github.ambry.rest.RestResponseChannel;
+import com.github.ambry.rest.RestServiceErrorCode;
+import com.github.ambry.rest.RestServiceException;
+import com.github.ambry.rest.RestUtils;
+import com.github.ambry.router.ReadableStreamChannel;
+import java.nio.ByteBuffer;
+import java.util.GregorianCalendar;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.github.ambry.frontend.FrontendUtils.*;
+
+
+/**
+ * Handle requests to get dataset using {@link AccountService}.
+ */
+public class GetDatasetsHandler {
+  private static final Logger LOGGER = LoggerFactory.getLogger(GetDatasetsHandler.class);
+
+  private final SecurityService securityService;
+  private final AccountService accountService;
+  private final FrontendMetrics frontendMetrics;
+  private final AccountAndContainerInjector accountAndContainerInjector;
+
+  /**
+   * Constructs a handler for handling requests for getting dataset.
+   * @param securityService the {@link SecurityService} to use.
+   * @param accountService the {@link AccountService} to use.
+   * @param frontendMetrics {@link FrontendMetrics} instance where metrics should be recorded.
+   * @param accountAndContainerInjector helper to resolve account and container for a given request.
+   */
+  GetDatasetsHandler(SecurityService securityService, AccountService accountService, FrontendMetrics frontendMetrics,
+      AccountAndContainerInjector accountAndContainerInjector) {
+    this.securityService = securityService;
+    this.accountService = accountService;
+    this.frontendMetrics = frontendMetrics;
+    this.accountAndContainerInjector = accountAndContainerInjector;
+  }
+
+  /**
+   * Asynchronously get account metadata.
+   * @param restRequest the {@link RestRequest} that contains the request parameters and body.
+   * @param restResponseChannel the {@link RestResponseChannel} where headers should be set.
+   * @param callback the {@link Callback} to invoke when the response is ready (or if there is an exception).
+   */
+  void handle(RestRequest restRequest, RestResponseChannel restResponseChannel,
+      Callback<ReadableStreamChannel> callback) throws RestServiceException {
+    RestRequestMetrics requestMetrics =
+        frontendMetrics.getDatasetsMetricsGroup.getRestRequestMetrics(restRequest.isSslUsed(), false);
+    restRequest.getMetricsTracker().injectMetrics(requestMetrics);
+    // get dataset request have their account/container name in request header, so checks can be done at early stage.
+    accountAndContainerInjector.injectAccountAndContainerForPostRequest(restRequest,
```

Comment:
the function name is a bit confusing. It seems like this injector function was designed for Post request.

Reply:
Updated.